drivers/ethernet/eth_e1000.c

```diff
@@ -134,22 +134,19 @@ static void e1000_isr(struct device *device)
 
 int e1000_probe(struct device *device)
 {
-	struct e1000_dev *dev = device->driver_data;
-
-	pci_bus_scan_init();
-
-	if (pci_bus_scan(&dev->pci)) {
-
-		pci_enable_regs(&dev->pci);
+	const pcie_bdf_t bdf = PCIE_BDF(0, 3, 0);
+	const u32_t flags = PCIE_CONF_CMDSTAT_MEM | PCIE_CONF_CMDSTAT_MASTER;
 
-		pci_enable_bus_master(&dev->pci);
-
-		pci_show(&dev->pci);
+	struct e1000_dev *dev = device->driver_data;
+	int retval = -ENODEV;
 
-		return 0;
+	if (pcie_probe(PCIE_BDF(0, 3, 0), PCIE_ID(0x8086, 0x100e))) {
```

Comment:
> Earlier we could discover IRQ from struct pci_dev_info. Is there way to get this with the new PCIE?

Actually, the mechanism to do this was pretty broken, and only worked properly on limited platforms.  The problem is that the "right" way to map PCI interrupts on many platforms (read: x86, the only PCI platform we currently support) is to use ACPI, which is something we'd like to avoid.  We've been lucky in the past: on the galileo, for example, the hardware mechanism is known.  Ditto for QEMU.  On newer platforms we won't be so lucky.  It was in fact this precise problem that motivated the PCI code overhaul in the first place.

>I planned to add support for dynamic IRQ registration in particularly for simics and also for qemu with non-default e1000 instantiation from the command line.

The good news is that the new PCIe code supports MSI, so on 90% of the PCI and *all* PCIe devices we're likely to encounter, we can simply tell the device which interrupts to trigger, rather than trying to figure out how to map the traces on the board to interrupt controllers and such.

QEMU supports MSI, though I'm not sure if that extends to the e1000 driver (yet).

Reply:
>  would it be a good idea not to restrict the scan to particular bus/device combination?

Scanning the bus would not be the typical use case.  It can of course be done (see `drivers/pcie/shell.c` which enumerates the bus for debugging purposes), but there is no generic mechanism to do this.  The general device model in Zephyr assumes a fixed set of devices in fixed locations; at present, those "fixed locations" are MMIO and IRQs.  If the device is PCI(e), then it is still in a fixed location.  The only difference is that that fixed location is a bus/device/function.